tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedErrorMatcher.cs

```diff
@@ -0,0 +1,135 @@
+            var mongoCommandException = actualException as MongoCommandException;
+
+            mongoCommandException.Code.Should().Be(expectedErrorCode);
+        }
+
+        private void AssertErrorCodeName(Exception actualException, string expectedErrorCodeName)
+        {
+            actualException.Should().BeOfType<MongoCommandException>();
+            var mongoCommandException = actualException as MongoCommandException;
+
+            mongoCommandException.CodeName.Should().Be(expectedErrorCodeName);
+        }
+
+        private void AssertErrorContains(Exception actualException, string expectedSubstring)
+        {
+            actualException.Message.Should().ContainEquivalentOf(expectedSubstring);
+        }
+
+        private void AssertErrorLabelsContain(Exception actualException, IEnumerable<string> expectedErrorLabels)
+        {
+            actualException.Should().BeAssignableTo<MongoException>();
+            var mongoException = actualException as MongoException;
+
+            mongoException.ErrorLabels.Should().Contain(expectedErrorLabels);
+        }
+
+        private void AssertErrorLabelsOmit(Exception actualException, IEnumerable<string> expectedAbsentErrorLabels)
+        {
+            actualException.Should().BeAssignableTo<MongoException>();
+            var mongoException = actualException as MongoException;
+
+            mongoException.ErrorLabels.Should().NotContain(expectedAbsentErrorLabels);
+        }
+
+        private void AssertExpectResult(Exception actualException, BsonDocument expectedResult)
+        {
+            actualException.Should().BeOfType<MongoBulkWriteException<BsonDocument>>();
+            var bulkWriteException = actualException as MongoBulkWriteException<BsonDocument>;
+            var bulkWriteResult = bulkWriteException.Result;
+            var actualResult = new BsonDocument
+            {
+                { "deletedCount", bulkWriteResult.DeletedCount },
+                { "insertedCount", bulkWriteResult.InsertedCount },
+                { "matchedCount", bulkWriteResult.MatchedCount },
+                { "modifiedCount", bulkWriteResult.ModifiedCount },
+                { "upsertedCount", bulkWriteResult.Upserts.Count },
+                { "upsertedIds", new BsonDocument(bulkWriteResult.Upserts.Select(x => new BsonElement(x.Index.ToString(), x.Id))) }
+            };
+
+            actualResult.Should().Be(expectedResult);
+        }
+
+        private void AssertIsClientError(Exception actualException, bool expectedIsClientError)
+        {
+            var actualIsClientError =
+                actualException is MongoClientException ||
+                actualException is BsonException;
+
+            actualIsClientError.Should().Be(expectedIsClientError);
+        }
+
+        private void AssertIsError(Exception actualException, bool expectedIsError)
+        {
+            expectedIsError.Should().BeTrue("Test files MUST NOT specify false.");
```

Comment:
Don't use `Should` here... this is a problem with the test file not a failed test:

    if (expectedIsError == false)
    {
        throw new FormatException("expectedIsError is false.");
    }